packages/gatsby-plugin-guess-js/src/gatsby-browser.js

```diff
@@ -69,6 +73,8 @@ exports.onPrefetchPathname = ({ pathPrefix }, pluginOptions) => {
           resources.push(`static/d/${___dataPaths[page.jsonName]}.json`)
           // TODO add support for pathPrefix
           resources.forEach(r => prefetch(`/${r}`))
+
+          onPostPrefetchPathname({ pathname: p })
```

Comment:
Could we refactor this a bit so `loader.js` exposes `prefetchResourcesForPath` function and `guess` plugin and default gatsby prefetching use same implementation (and `onPostPrefetchPathname` api is called just from `loader.js`) ?